### DISCUSSION TITLE
Request - Your help in improving the default system prompts

### DISCUSSION THREAD

**Discussion by olimorris** [olimorris/codecompanion.nvim#80] [Announcements]
Hello All 👋🏼 

In my quest to make this plugin yield the best outputs from OpenAI, Anthropic AND Ollama, I'm seeking your feedback in how we can tweak the default `system_prompts` (and `system` prompts in the `default_prompts` table) in the [config.lua](https://github.com/olimorris/codecompanion.nvim/blob/main/lua/codecompanion/config.lua) file.

Ultimately, I'm trying to get to the level of response that you may observe in GitHub's Copilot Chat in VS Code. Helpful, but overly terse and low in token consumption.

As an aside; the changes I've made to the plugin in recent weeks should allow for Gemini integration. It's low on my todo list but conscious they're silently making really interesting improvements to their models.

**Comment by JPyke3** [comment]
Hmmmm... Do you think it's worth trying to find the best output across the board? Just from what I have used at least it seems almost always beneficial to tweak system prompts based on the specific AI model. I guess we need to start somewhere but maybe it's a consideration to at least split `OpenAI` and `Anthropic`'s models at a minimum. Unfortunately due the nature of `Ollama` it's obviously hard as a user can run whatever they want, so that's maybe *less* important.

Thought I would throw my hat in the ring anyways, I will spend this WE refining my Ollama prompts and post them up so people can take a look

**Reply by olimorris** [reply]
Very good suggestion. Perhaps even if it's not at a model level we could easily make it an adapter one for Anthropic and OpenAI. I'll play around in the coming days.

Your Ollama prompts would be very helpful.

**Reply by JPyke3** [reply]
Yeah I think at an adapter level seems pretty reasonable. Now that I think about it, I am running some `gpt-turbo-3.5` prompts with `gpt-4o` and they've never had any issues. So I think that's a good way to run it
